rvgo/fast/elf.go

```diff
@@ -69,6 +69,16 @@ func PatchVM(f *elf.File, vmState *VMState) error {
 			"runtime.main.func1",        // patch out: main.func() { newm(sysmon, ....) }
 			"runtime.deductSweepCredit", // uses floating point nums and interacts with gc we disabled
 			"runtime.(*gcControllerState).commit",
+			// these prometheus packages rely on concurrent background things. We cannot run those.
+			"github.com/prometheus/client_golang/prometheus.init",
+			"github.com/prometheus/client_golang/prometheus.init.0",
+			"github.com/prometheus/procfs.init",
+			"github.com/prometheus/common/model.init",
+			"github.com/prometheus/client_model/go.init",
+			"github.com/prometheus/client_model/go.init.0",
+			"github.com/prometheus/client_model/go.init.1",
+			// skip flag pkg init, we need to debug arg-processing more to see why this fails
```

Comment:
curious to know what the failure is. Does the program crash during pkg init?

Reply:
Honestly, these lines are copied from [Cannon](https://github.com/ethereum-optimism/optimism/blob/v1.7.3/cannon/mipsevm/patch.go#L76) 😅
So I have no idea about the details..